File: src/main/java/me/brendn/oak/mcbeta/BetaAdapter.java
package me.brendn.oak.mcbeta;

import me.brendn.oak.api.common.entity.Entity;
import me.brendn.oak.api.common.world.World;
import me.brendn.oak.api.event.EventManager;
import me.brendn.oak.mcbeta.gui.BetaTextRenderer;
import me.brendn.oak.api.MinecraftAdapter;
import me.brendn.oak.api.client.gui.TextRenderer;
import net.minecraft.client.Minecraft;

import java.io.File;

/**
 * Beta 1.7.3 Implementation of the {@link MinecraftAdapter}
 */
public class BetaAdapter implements MinecraftAdapter {

	private Minecraft mc;

	private World world;
	private BetaTextRenderer textRenderer;

	public BetaAdapter(Minecraft mc) {
		this.mc = mc;
		this.textRenderer = new BetaTextRenderer(mc);
	}

	@Override
	public File getDirectory() {
		return Minecraft.getMinecraftDir();
	}

	@Override
	public TextRenderer getTextRenderer() {
		return this.textRenderer;
	}

	@Override
	public EventManager getEventManager() {
		return null;
	}

	@Override
	public String getName() {
		return "BetaOak";
	}

	@Override
	public String getMinecraftVersion() {
		return "b1.7.3";
	}

	@Override
	public World getWorld() {
		return world;
	}

	@Override
	public void setWorld(World world) {
		this.world = world;
		debugLog("Loaded world!");
		debugLog("World seed: " + world.getSeed());
		debugLog("Loaded entities: " + world.getLoadedEntities().size());
	}

	private void debugLog(String s) {
		System.out.println("[API/DEBUG]: " + s);
	}
}

File: src/main/kotlin/me/brendn/oak/mcbeta/BetaAdapter.kt
package me.brendn.oak.mcbeta

import me.brendn.oak.api.MinecraftAdapter
import me.brendn.oak.api.client.entity.ClientPlayer
import me.brendn.oak.api.client.gui.TextRenderer
import me.brendn.oak.api.common.entity.living.animal.Pig
import me.brendn.oak.api.common.world.World
import me.brendn.oak.api.event.EventManager
import me.brendn.oak.mcbeta.gui.BetaTextRenderer
import net.minecraft.client.Minecraft
import org.apache.logging.log4j.Level
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.io.File

class BetaAdapter(val mc: Minecraft) : MinecraftAdapter {

	private val textRenderer: TextRenderer = BetaTextRenderer(mc)
	private lateinit var world: World

	override fun getDirectory(): File = Minecraft.getMinecraftDir()

	override fun getTextRenderer() = textRenderer

	override fun getEventManager(): EventManager {
		TODO("not implemented")
	}

	override fun getMinecraftVersion() = "b1.7.3"

	override fun getWorld() = world

	override fun setWorld(world: World) {
		getLogger().log(Level.INFO, "Loaded world!")
		getLogger().log(Level.INFO, "World Seed: ${world.getSeed()}")
		getLogger().log(Level.INFO, "Loaded entities: ${world.getLoadedEntities().size}")

		if (world.getLoadedEntities().isNotEmpty()) {
			world.getLoadedEntities().stream().filter { it is Pig }.forEach {
				getLogger().log(Level.INFO, "Pig located at ${it.getLocation()}")
			}

			val loc = world.getLoadedEntities().first().getLocation()
			val biome = world.getBiome(loc.x.toInt(), loc.z.toInt())
			getLogger().log(Level.INFO, "Biome at $loc: ${biome.getName()}")
		}

		if (getPlayer() != null) {
			val player = getPlayer()!!
			val loc = player.getLocation()
			val eID = player.getID()
			val name = player.getUsername()

			getLogger().log(Level.INFO, "Player $name (#$eID) spawned at $loc")
		}
	}

	override fun getLogger(): Logger = LogManager.getLogger("Oak")

	override fun getPlayer(): ClientPlayer? = mc.thePlayer as ClientPlayer
}